base/linalg/symmetric.jl

```diff
@@ -107,6 +107,24 @@ end
     r
 end
 
+function setindex!(A::Symmetric, v, i::Integer, j::Integer)
+    if (A.uplo == 'U' && i >= j) || (A.uplo == 'L' && i <= j)
```

Comment:
That makes sense. Then I'm back to defining specific methods for things, since only being able to set the diagonal doesn't allow more general operations, but maybe that's fine.